### PR TITLE
#2475 Command palette with better actions ux flow for query editor

### LINKED ISSUE #2475: Add button to open the query view editor's command palette
The command palette for the query view can be brought up by typing _F1_. That is the default hotkey for the Ace editor that we are using. 

As users are not familiar with ACE hotkeys, we should add a small button somewhere that will trigger the command palette to open. This will also help the users find other useful hotkeys as the hotkeys are shown along with the commands.

### PR DESCRIPTION
**Pull request by matusmlichsk**
[//]: # (Request Template Witsml Explorer)

[//]: # (Thank you for contributing to WITSML Explorer! Before submitting this PR, please fill in the following template.)

## Fixes

[//]: # (Write the GitHub issue number starting with #, or, for Equinor only, the Jira issue number starting with WE-)

This pull request fixes #2475 

## Description

[//]: # (Please include a written summary of the changes and which issue is fixed. List any dependencies that are required for this change._)

 Command palette with better actions ux flow for query editor.

## Type of change

[//]: # (Mark any of the types of change that apply.)

* [ ] Bugfix
* [ ] New feature (non-breaking change which adds functionality)
* [x] Enhancement of existing functionality
* [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
* [ ] This change requires a documentation update

## Impacted Areas in Application

[//]: # (List general components of the application that this PR will affect)

* [x] Frontend
* [ ] API
* [ ] WITSML
* [ ] Desktop
* [ ] Other (please describe)

## Checklist:

[//]: # (Please tick all the boxes or remove the ones that aren't needed and explain why)

*Communication*
* [ ] I have made corresponding changes to the documentation
* [ ] PR affects application security

*Code quality*
* [x] I have self-reviewed my code
* [x] No new warnings are generated

*Test coverage*
* [ ] New code is covered by passing tests

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

### CODE REVIEW

File: Src/WitsmlExplorer.Frontend/components/QueryEditor.tsx [hunk 1]
```diff
@@ -12,56 +12,85 @@ import AceEditor from "react-ace";
...
 
-  const onLoad = (editor: any) => {
+  const onLoadInternal = (editor: any) => {
     editor.renderer.setPadding(10);
     editor.renderer.setScrollMargin(10);
     if (readonly) {
       editor.renderer.$cursorLayer.element.style.display = "none";
     } else {
       editor.completers = [customCompleter];
+      editor.renderer.on("afterRender", (_: any, renderer: any) => {
```
Comment (eliasbruvik) [inline Src/WitsmlExplorer.Frontend/components/QueryEditor.tsx]: Move the _afterRender_ callback out of the _readonly_ condition. After my previous update, widgets are added even for the readonly editor.

**Review by eliasbruvik — CHANGES_REQUESTED** [review]

File: Src/WitsmlExplorer.Frontend/components/QueryEditor.tsx (hunk 1 above)
Comment (matusmlichsk) [reply Src/WitsmlExplorer.Frontend/components/QueryEditor.tsx]: Okay, makes sense now

**Review by eliasbruvik — CHANGES_REQUESTED** [review]
Got a request from Sebastian to move the execute button to the left part as the right part is just the result.

**Comment by matusmlichsk** [comment]
redesigned and moved to other actions 
image

**Review by eliasbruvik — CHANGES_REQUESTED** [review]
Looks good, just noticed that you can toggle the options all the way over by the execute button. Can you reduce the width a bit?

image

**Comment by matusmlichsk** [comment]
Will do, but this is native element behavior due better accessibility in browser.

**Review by eliasbruvik — APPROVED** [review]
*Reviewer checklist*
- [x] Read issue and PR
- [x] Pulled branch and manually tested
- [x] Verify that PR resolves issue
- [x] Reviewed the code

**Pull request merged**